src/resources/js/components/AccountCreation.vue

```diff
@@ -1,50 +1,124 @@
+      </div>
+      <div class="form-group">
+        <input type="email" class="input" name="email" v-model="input.email" placeholder="email" />
+      </div>
+      <div class="form-group">
+        <input type="password" class="input" name="password" v-model="input.password" placeholder="Mot de passe" />
+      </div>
+      <div class="form-group">
+        <input type="password" class="input" name="passwordConfirm" v-model="input.passwordConfirm" placeholder="Confirmer mot de passe" />
+      </div>
+      <div class="form-group">
+        <button type="button" class="button" v-on:click="register()">Créer compte</button>
+      </div>
+      
+      <p v-if="isBlank"> Manque des informations</p>
+      <p v-if="isIncorect"> mot de passe différent</p>
+    </form>
+  </div>
 </template>
 
 <script>
+    import axios from 'axios'
+    import router from '../routes'
+
     export default {
-        data() {
-            return {
-                greetings: 'Please fill the followings',
-                firstname: null,
-                lastname: null,
+    name : 'Register',
+    data () {
+        return{
+          input: {
+              username: '',
+              email: '',
+              password: '',
+              passwordConfirm: ''
+          },
+      darkBlue: '#5F74E8',
+      lightBlue: '#69A7FF',
+      isBlank: false,
+      isIncorect: false}
+
+    },
+    methods: {
+      register() {
+         if (this.isFieldsEmpty()){ //if not blank
+            if (this.isPasswordConfirmMatchPassword()){ //if password is different of comfirm password
+                //sends credentials to backend 
+                axios.post(
+                '/register',{
+                    name: this.username,
+                    email: this.email,
+                    password: this.password
+                })
+                .then(function (data){
+                // this function is launched after the post request
+                console.log(data);
+                })
+                .catch(function (error){
+                console.log(error);
+                });
+
+                this.$router.replace({name: 'login'});
+            } else {
+                this.isIncorect = true;
             }
-            
-        },
-        methods: {
-                OnSubmit () {
-                    let newAccount = {
-                        firstname: this.firstname,
-                        lastname: this.lastname
-                    }
-                }
-        }
-    };
-</script>
\ No newline at end of file
+         } else {
+           this.isBlank=true;
+         }
+      },
+      //check if both password are the same
+      isPasswordConfirmMatchPassword(){
+        return (this.input.password === this.input.passwordConfirm);
+      },
+      //check if an input is empty
+      isFieldsEmpty(){
+        return (this.input.username != "" && this.input.email != "" && this.input.password != "" && this.input.passwordConfirm != "");
```

Comment:
Instead of using a function you can use the [required](https://www.w3schools.com/tags/att_input_required.asp) element of HTML5, this will handle the job. Or you can output which text input must be fill

Reply:
the required attribute of HTML5 isn't enough to validate data, you still need to have JS checks

Reply:
@kayoumido  you're right but in this case the required attribute is enough to check that a field is not empty but yes for further more complex checks it's nice to have JS checks